lib/bugsnag/reporter.ex

```diff
@@ -86,6 +86,6 @@ defmodule Bugsnag.Reporter do
   end
 
   defp add_stacktrace(options) do
-    if options[:stacktrace], do: options, else: put_in(options[:stacktrace], System.stacktrace())
+    if options[:stacktrace], do: options, else: put_in(options[:stacktrace], __STACKTRACE__)
```

Comment:
the stacktrace needs to be passed explicitly from the point where the exception occurs, i guess